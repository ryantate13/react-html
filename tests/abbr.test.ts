import {abbr} from '../src';

describe('abbr', () => {
    it('renders with children', () => {
        expect(abbr({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(abbr({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(abbr()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(abbr({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(abbr({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(abbr({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(abbr({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(abbr({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(abbr({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(abbr({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(abbr({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(abbr({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(abbr({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(abbr({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(abbr({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(abbr({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(abbr({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(abbr({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(abbr({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(abbr({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(abbr({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(abbr({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(abbr({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(abbr({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(abbr({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(abbr({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(abbr({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(abbr({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(abbr({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(abbr({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(abbr({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(abbr({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(abbr({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(abbr({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(abbr({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(abbr({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(abbr({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(abbr({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(abbr({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(abbr({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(abbr({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(abbr({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(abbr({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(abbr({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(abbr({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(abbr({translate: "no"})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(abbr({title: "TEST"})).toMatchSnapshot();
    });
});