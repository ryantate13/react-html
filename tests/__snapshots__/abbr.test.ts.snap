// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`abbr renders with accessKey="a,b,c,d" 1`] = `
<abbr
  accessKey="a,b,c,d"
/>
`;

exports[`abbr renders with autoCapitalize="characters" 1`] = `
<abbr
  autoCapitalize="characters"
/>
`;

exports[`abbr renders with autoCapitalize="none" 1`] = `
<abbr
  autoCapitalize="none"
/>
`;

exports[`abbr renders with autoCapitalize="off" 1`] = `
<abbr
  autoCapitalize="off"
/>
`;

exports[`abbr renders with autoCapitalize="on" 1`] = `
<abbr
  autoCapitalize="on"
/>
`;

exports[`abbr renders with autoCapitalize="sentences" 1`] = `
<abbr
  autoCapitalize="sentences"
/>
`;

exports[`abbr renders with autoCapitalize="words" 1`] = `
<abbr
  autoCapitalize="words"
/>
`;

exports[`abbr renders with children 1`] = `
<abbr>
  TEST
</abbr>
`;

exports[`abbr renders with className="foo bar baz" 1`] = `
<abbr
  className="foo bar baz"
/>
`;

exports[`abbr renders with contentEditable="false" 1`] = `
<abbr
  contentEditable="false"
/>
`;

exports[`abbr renders with contentEditable="true" 1`] = `
<abbr
  contentEditable="true"
/>
`;

exports[`abbr renders with dir="auto" 1`] = `
<abbr
  dir="auto"
/>
`;

exports[`abbr renders with dir="ltr" 1`] = `
<abbr
  dir="ltr"
/>
`;

exports[`abbr renders with dir="rtl" 1`] = `
<abbr
  dir="rtl"
/>
`;

exports[`abbr renders with draggable="false" 1`] = `
<abbr
  draggable="false"
/>
`;

exports[`abbr renders with draggable="true" 1`] = `
<abbr
  draggable="true"
/>
`;

exports[`abbr renders with empty attributes 1`] = `<abbr />`;

exports[`abbr renders with hidden="false" 1`] = `
<abbr
  hidden={false}
/>
`;

exports[`abbr renders with hidden="true" 1`] = `
<abbr
  hidden={true}
/>
`;

exports[`abbr renders with id="TEST" 1`] = `
<abbr
  id="TEST"
/>
`;

exports[`abbr renders with inputMode="decimal" 1`] = `
<abbr
  inputMode="decimal"
/>
`;

exports[`abbr renders with inputMode="email" 1`] = `
<abbr
  inputMode="email"
/>
`;

exports[`abbr renders with inputMode="none" 1`] = `
<abbr
  inputMode="none"
/>
`;

exports[`abbr renders with inputMode="numeric" 1`] = `
<abbr
  inputMode="numeric"
/>
`;

exports[`abbr renders with inputMode="search" 1`] = `
<abbr
  inputMode="search"
/>
`;

exports[`abbr renders with inputMode="tel" 1`] = `
<abbr
  inputMode="tel"
/>
`;

exports[`abbr renders with inputMode="text" 1`] = `
<abbr
  inputMode="text"
/>
`;

exports[`abbr renders with inputMode="url" 1`] = `
<abbr
  inputMode="url"
/>
`;

exports[`abbr renders with is="foo" 1`] = `
<abbr
  is="foo"
/>
`;

exports[`abbr renders with itemID="https://google.com/" 1`] = `
<abbr
  itemID="https://google.com/"
/>
`;

exports[`abbr renders with itemProp="https://google.com/" 1`] = `
<abbr
  itemProp="https://google.com/"
/>
`;

exports[`abbr renders with itemRef="foo bar" 1`] = `
<abbr
  itemRef="foo bar"
/>
`;

exports[`abbr renders with itemScope="false" 1`] = `
<abbr
  itemScope={false}
/>
`;

exports[`abbr renders with itemScope="true" 1`] = `
<abbr
  itemScope={true}
/>
`;

exports[`abbr renders with itemType="https://google.com/" 1`] = `
<abbr
  itemType="https://google.com/"
/>
`;

exports[`abbr renders with lang="" 1`] = `
<abbr
  lang=""
/>
`;

exports[`abbr renders with lang="es-419" 1`] = `
<abbr
  lang="es-419"
/>
`;

exports[`abbr renders with slot="TEST" 1`] = `
<abbr
  slot="TEST"
/>
`;

exports[`abbr renders with spellCheck="false" 1`] = `
<abbr
  spellCheck="false"
/>
`;

exports[`abbr renders with spellCheck="true" 1`] = `
<abbr
  spellCheck="true"
/>
`;

exports[`abbr renders with style="[object Object]" 1`] = `
<abbr
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`abbr renders with tabIndex="1" 1`] = `
<abbr
  tabIndex={1}
/>
`;

exports[`abbr renders with title="TEST" 1`] = `
<abbr
  title="TEST"
/>
`;

exports[`abbr renders with title="TEST" 2`] = `
<abbr
  title="TEST"
/>
`;

exports[`abbr renders with translate="no" 1`] = `
<abbr
  translate="no"
/>
`;

exports[`abbr renders with translate="yes" 1`] = `
<abbr
  translate="yes"
/>
`;

exports[`abbr renders without attributes 1`] = `<abbr />`;
